src/hotspot/share/opto/multnode.hpp

```diff
@@ -100,6 +100,10 @@ class ProjNode : public Node {
   //                                             other_proj->[region->..]call_uct"
   // null otherwise
   CallStaticJavaNode* is_uncommon_trap_if_pattern(Deoptimization::DeoptReason reason = Deoptimization::Reason_none) const;
+  // Return true if this projection doesn't end with an uncommon trap but, even though several cfg paths are branching out
+  // from here, they all end with an uncommon trap
```

Comment:
This comment is a bit confusing.
`Return true if this projection doesn't end with an uncommon trap`
Sounds like if you find no uncommon trap you return true.

Suggestion:
Check if all cfg paths lead to some (possibly multiple different) uncommon trap or Halt node.
Traverse Region, If, IfProj nodes.

Control question: a Halt node is also an uncommon trap in your definition then?